osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SecretManagementService.java

```diff
@@ -1,181 +1,317 @@
+        return this.aesEncryptKey(key);//, keyType);
 
-    @Override
-    public byte[] getDlmsMasterKey(String deviceIdentification) {
-        throw new NotImplementedException();
     }
 
     @Override
-    public byte[] getDlmsAuthenticationKey(String deviceIdentification) {
+    public byte[] rsaDecrypt(final byte[] externallyEncryptedKey) //, final SecurityKeyType keyType)
+            throws FunctionalException {
 
         try {
-            GetSecretsRequest request = getSoapRequestForKey(deviceIdentification,
-                    SecretType.E_METER_AUTHENTICATION_KEY);
-            GetSecretsResponse response = secretManagementClient.getSecretsRequest(request);
-            Optional<TypedSecret> optionalTypedSecret = getTypedSecretFromSoapResponse(response,
-                    SecretType.E_METER_AUTHENTICATION_KEY);
+            return this.rsaEncryptionService.decrypt(externallyEncryptedKey);
+        } catch (final Exception e) {
+            LOGGER.error("Unexpected exception during decryption", e);
+            throw new FunctionalException(FunctionalExceptionType.DECRYPTION_EXCEPTION, ComponentType.PROTOCOL_DLMS, e);
+            // new EncrypterException(
+            //         String.format("Unexpected exception during decryption of key.", keyType)));
+        }
 
-            byte[] decryptedKey = decryptSoapSecret(deviceIdentification, optionalTypedSecret.orElseThrow(
-                    () -> new IllegalStateException("Secret not found:" + deviceIdentification)));
+    }
 
-            log.trace("DlmsAuthenticationKey for device " + deviceIdentification + " is " + Hex.encodeHexString(decryptedKey));
+    //@Override
+    public byte[] aesEncryptKey(final byte[] key) /*, final SecurityKeyType keyType)*/ throws FunctionalException {
 
-            return decryptedKey;
-        } catch (Exception e) {
-            throw new EncrypterException("Error while retrieving authentication key", e);
+        try {
+            return this.aesEncryptionService.encrypt(key);
+        } catch (final Exception e) {
+            LOGGER.error("Unexpected exception during encryption", e);
+            //TODO refactor this?
+            throw new FunctionalException(FunctionalExceptionType.ENCRYPTION_EXCEPTION, ComponentType.PROTOCOL_DLMS, e);
+            //   new EncrypterException(
+            //           String.format("Unexpected exception during encryption of %s key.", keyType)));
         }
     }
 
+    /*
+     * Decrypts the given symmetrically encrypted key.
+     * <p>
+     * <strong>NB:</strong> Only decrypt keys like this at the moment they are
+     * required as part of the communication with a device.
+     *
+     * @param encryptedKey
+     *         key encrypted with the symmetrical key internal to the DLMS
+     *         protocol adapter.
+     *
+     * @return the plain key, or an empty byte array if
+     *         {@code encryptedKey == null}
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)